CMakeLists.txt

```diff
@@ -166,9 +166,9 @@ if (NJOBS)
   set(CMAKE_JOB_POOL_PRECOMPILE_HEADER custom)
 endif()
 
-# GCC 8.1 is required to support <charconv> header inclusion in base/string_utils.hpp, otherwise 7.0 is sufficient
-if (CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 8.1)
-  message(FATAL_ERROR "Minimum supported g++ version is 8.1 yours is ${CMAKE_CXX_COMPILER_VERSION}")
+# GCC 8.1 is required to support <charconv> header inclusion in base/string_utils.hpp
+if (CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 14.0)
+  message(FATAL_ERROR "Minimum supported g++ version is 14.0, yours is ${CMAKE_CXX_COMPILER_VERSION}")
```

Comment:
this is correct, right?

Reply:
No, please check [this table](https://gcc.gnu.org/projects/cxx-status.html), and it would be great to also test on different GCC versions. Highly likely that gcc-10 or 11 would be ok. You can experiment locally in this PR by adding other gcc version matrix values in the .github workflow check.

Reply:
checked, gcc-9 fails but gcc-10 works, have updated PR 👍️